### ISSUE TITLE
Organizing Theme Better

### ISSUE THREAD

**Issue by dmlb2000** [pacifica/pacifica-main-site#34]
The CSS is mostly centralized in `index.scss` it would be nice to have a discussion about organizing theme in a site like this. Every time I mess with CSS on a site I always feel like I'm constantly getting tripped up while I'm trying to sprint to a finish.

 * What is a good strategy to organize theme in a React site like this?
 * What are appropriate levels of breakdown to organize the theme and not go crazy later?

Thoughts?

**Comment by thurber** [comment]
I definitely know what you mean; I've yet to find a strategy that I'm totally satisfied with. Lately, I've been making use of `styled-components` to unify the react components with their styling. Styled components allow the use of javascript and component props in the styling, with a centralized theme file from which to import common styles (colors, spacing, etc).

For a `sass` based site, the general strategy I've seen is to have an `.scss` file per react component file, with only a very minimum of global styling defined in an `index.scss`. In this paradigm, each component should have a root class prefix to prevent collisions, i.e.:

Header.js
```js
import React from 'react';
import 'Header.scss';

function Header({ ...props }) {
  return (
    <div className='Header-wrapper'>
      <div className='Header-item'>
         Hello world!
      </div>
    </div>
  );
}
```

Header.scss
```scss
@import 'theme' 
.Header-wrapper {
  display: flex;
}

.Header-item {
  color: $themeColor;
}
```

It get's harder when using something like Bootstrap but still trying to implement a fairly custom theme, since Bootstrap has endless cascades of styles applied before ever getting to one's own customizations. This site is already pretty far along, so unless there's going to be a lot of follow-on work later, probably best just to deal with the overrides as needed.

**Comment by dmlb2000** [comment]
@thurber this makes some sense, I'll keep this in mind as we go forward.

**Comment by dmlb2000** [comment]
I'm satisfied closing this, fixed in #35
